Sources/GenerateValFileTests/GenerateValFileTests.swift

```diff
@@ -0,0 +1,153 @@
+import ArgumentParser
+import Foundation
+
+/// A command-line tool that, given a list of .val files, generates XCTest cases for annotated .val
+/// files as part of our build process.
+@main
+struct GenerateValFileTests: ParsableCommand {
+
+  @Option(
+    name: [.customShort("o")],
+    help: ArgumentHelp("Write output to <file>.", valueName: "output-swift-file"),
+    transform: URL.init(fileURLWithPath:))
+  var outputURL: URL
+
+  @Option(
+    name: [.customShort("n")],
+    help: ArgumentHelp("Name of generated test case.", valueName: "XCTestCase name"))
+  var testCaseName: String
+
+  @Argument(
+    help: "Paths of annotated val source files to be tested.",
+    transform: URL.init(fileURLWithPath:)
+  )
+  var valSourceFiles: [URL]
+
+  /// Returns the Swift source of the test function for the Val at `source`, having `firstLine` as
+  /// its first line.
+  func swiftFunctionTesting(valAt source: URL, withFirstLine firstLine: Substring) throws -> String
```

Comment:
```suggestion
  func swiftFunctionTesting(_ source: URL, withFirstLine firstLine: Substring) throws -> String
```
Not convinced the label is useful.

Reply:
I am.  This is a case of weak type information.  We could be testing anything.

Reply:
Works for me then.